### ISSUE TITLE
Read source code from StdInput

### ISSUE THREAD

**Issue by Sweetchuck** [phpmd/phpmd#542]
It would be very useful to validate the code in a Git pre-commit hook.
Something like this:
`git show :foo.php | phpmd ...`
  
I tried the following, but the exit code was `0` without any output with a PHP file which contains several errors.
`git show :foo.php | bin/phpmd php://stdin text naming`

The problem is that the `\PDepend\Engine::createFileIterator()` iterates over the `$this->files` which contains a file path like this `php://stdin`, but that file path is added to the `$fileIterator` like this:
`$fileIterator->append(new Iterator(new \GlobIterator($file), $this->fileFilter));`,
and later the `\PDepend\Input\Iterator::getFullPath()` is called, which calls the `::getRealPath()` method of the current item, which is an instance of `\SplFileObject`, and it always returns false.

In short this is the problem:
```php
$item = new \SplFileObject('php://stdin');
$path = $item->getRealPath(); // $path === false
```

pdepend/pdepend#361

**Comment by kylekatarnls** [comment]
Note than the  alias `-` can be used instead of `php://stdin`:
```
echo ... | bin/phpmd - text naming
```

And the support of `php://stdin` should be fixed in the next release of PDepend.
